File: backend/app/schemas/inspection.py
# app/schemas/inspection.py
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

QAStatus = Literal["PASS", "WARNING", "CRITICAL_HAZARD"]
PhotoType = Literal["CASING", "TERMINAL_COVER", "DISPLAY", "TAMPER_SEAL"]


class InspectionTelemetryItem(BaseModel):
    id: str = Field(
        ..., description="Unique client inspection ID (e.g., ins_1726918195000)"
    )
    technician_id: str
    meter_serial_number: str
    timestamp_utc: str
    geo: tuple[float, float] = Field(..., description="[Latitude, Longitude]")
    qa_status: QAStatus
    hazard_reason: str | None = None
    kwh_reading: float | None = None
    ocr_confidence: float | None = None

    model_config = ConfigDict(from_attributes=True)


class Tier1BatchPayload(BaseModel):
    batch_id: str
    inspections: list[InspectionTelemetryItem]


class BatchIngestResponse(BaseModel):
    status: str
    batch_id: str
    ingested_count: int
    received_at: str


class ImageUploadResponse(BaseModel):
    status: str
    inspection_id: str
    photo_type: str
    path: str


class HealthResponse(BaseModel):
    status: str
    database: str
    timestamp_utc: str

File: src/core/types/inspection.ts
export type QAStatus = "PASS" | "WARNING" | "CRITICAL_HAZARD";
export type SyncStatus = "PENDING" | "SYNCING" | "SYNCED";

export interface InspectionRecord {
	id: string;
	technician_id: string;
	device_id: string;
	meter_serial_number: string;
	latitude: number;
	longitude: number;
	timestamp_utc: string;
	qa_status: QAStatus;
	hazard_reason: string | null;
	extracted_kwh: number | null;
	ocr_confidence: number | null;
	image_local_path: string;
	sync_tier1_status: SyncStatus;
	sync_tier2_status: SyncStatus;
	created_at?: string;
}

export interface Tier1BatchPayload {
	batch_id: string;
	inspections: Array<{
		id: string;
		technician_id: string;
		meter_serial_number: string;
		timestamp_utc: string;
		geo: [number, number];
		qa_status: QAStatus;
		hazard_reason: string | null;
		kwh_reading: number | null;
		ocr_confidence: number | null;
	}>;
}

export interface QueueMetrics {
	pendingTelemetry: number;
	syncedTelemetry: number;
	pendingImages: number;
}
